src/types.ts

```diff
@@ -116,7 +117,7 @@ export type SchemaAttribute = {
   default?: any;
   required?: boolean;
   matches?: SchemaMatches | ((config: Config) => SchemaMatches);
-  validate?(value: any, config: Config): ValidationError[];
+  validate?(value: any, config: Config, name: string): ValidationError[];
```

Comment:
I feel like we can avoid adding this new param by just encouraging folks to use currying. 

I have a vague recollection of adding this capability when I implemented the `CustomAttributeType` feature, but then removed it. 

I wonder if we should bind it to `this` instead? 

No matter what you decide, we should also pass it as the 3rd param of `CustomAttributeTypeInterface.validate`